libs/src/sdk/services/StorageService/StorageService.ts

```diff
@@ -0,0 +1,178 @@
+    return await this.uploadFile(file, onProgress, 'img_square')
+  }
+
+  async uploadProfilePicture(file: File, onProgress: ProgressCB = () => {}) {
+    return await this.uploadFile(file, onProgress, 'img_square')
+  }
+
+  async uploadCoverPhoto(file: File, onProgress: ProgressCB = () => {}) {
+    return await this.uploadFile(file, onProgress, 'img_backdrop')
+  }
+
+  async uploadFile(
+    file: File,
+    onProgress: ProgressCB,
+    template: 'audio' | 'img_square' | 'img_backdrop'
+  ) {
+    const formData = new FormData()
+    formData.append('template', template)
+    // TODO: Test this in a browser env
+    formData.append('files', isNodeFile(file) ? file.buffer : file, file.name)
+    const response = await this._makeRequest({
+      method: 'post',
+      url: '/mediorum/uploads',
+      data: formData,
+      headers: {
+        'Content-Type': `multipart/form-data; boundary=${formData.getBoundary()}`
+      },
+      onUploadProgress: (progressEvent) =>
+        onProgress(progressEvent.loaded, progressEvent.total)
+    })
+    return await this.pollProcessingStatus(
+      response.data[0].id,
+      template === 'audio'
+        ? MAX_TRACK_TRANSCODE_TIMEOUT
+        : MAX_IMAGE_RESIZE_TIMEOUT_MS
+    )
+  }
+
+  /**
+   * Works for both track transcode and image resize jobs
+   * @param id ID of the transcode/resize job
+   * @param maxPollingMs millis to stop polling and error if job is not done
+   * @returns successful job info, or throws error if job fails / times out
+   */
+  async pollProcessingStatus(id: string, maxPollingMs: number) {
+    const start = Date.now()
+    while (Date.now() - start < maxPollingMs) {
+      try {
+        const resp = await this.getProcessingStatus(id)
+        if (resp?.status === 'done') return resp
+        if (resp?.status === 'error') {
+          throw new Error(
+            `Upload failed: id=${id}, resp=${JSON.stringify(resp)}`
+          )
+        }
+      } catch (e) {
+        // Swallow errors caused by failure to establish connection to node so we can retry polling
+        console.error(`Failed to poll for processing status, ${e}`)
+      }
+
+      await wait(POLL_STATUS_INTERVAL)
+    }
+
+    throw new Error(`Upload took over ${maxPollingMs}ms. id=${id}`)
+  }
+
+  /**
+   * Gets the task progress given the task type and id associated with the job
+   * @param id the id of the transcoding or resizing job
+   * @returns the status, and the success or failed response if the job is complete
+   */
+  async getProcessingStatus(id: string) {
+    const { data } = await this._makeRequest({
+      method: 'get',
+      url: `/mediorum/uploads/${id}`
+    })
+    return data
+  }
+
+  /* ------- INTERNAL FUNCTIONS ------- */
+
+  /**
+   * Makes an axios request to the connected creator node
+   * @return response body
+   */
+  async _makeRequest(axiosRequestObj: AxiosRequestConfig) {
+    // TODO: This might want to have other error handling, request UUIDs, etc...
```

Comment:
Might make sense to make Linears for all the TODOs in this PR so we don't have to remember to come back to them, up to you though

Reply:
This one is theo's todo that I copied over from libs 😅 but yes will make sure we have items in linear for all the relevant tasks

Reply:
lolol / sgsg

Reply:
lol yeah this was because the v1 libs equivalent for `_makeRequest` had so much stuff and I think we don't need most of it. I'll definitely follow up to make sure it's robust with error handling and whatnot before prod tho!